java/core/src/java/org/apache/orc/impl/RecordReaderImpl.java

```diff
@@ -231,11 +224,9 @@ protected RecordReaderImpl(ReaderImpl fileReader,
     if (options.getPreFilterColumnNames() != null) {
       for (String colName : options.getPreFilterColumnNames()) {
         int expandColId = findColumns(evolution, colName);
+        // If the column is not present in the file then this can be ignored from read.
         if (expandColId != -1) {
           filterColIds.add(expandColId);
-        } else {
-          throw new IllegalArgumentException("Filter could not find column with name: " +
-              colName + " on " + evolution.getReaderBaseSchema());
```

Comment:
Could you confirm that what is the new error message? I understand that the thrown exception is the same, `IllegalArgumentException`, but I'm not sure about the information on the error message.

Reply:
Yeah, will add this wrapping into the find method as suggested by @pgaref